docs/item.rst

```diff
@@ -0,0 +1,93 @@
+item
+====
+
+.. dfhack-tool::
+    :summary: Perform bulk operations on items based on various properties.
+    :tags: fort productivity items
+
+Filter items in you fort by various properties (e.g., item type, material,
+wear-level, quality, ...), and perform the bulk operations forbid, dump, melt,
+and their inverses. By default, the tool does not consider artifacts. Outputs
+the number of items that matched the filters and were modified.
+
+Usage
+-----
+
+``item [ count | [un]forbid | [un]dump | [un]melt ] <filter options>``
```

Comment:
there have been requests on Reddit for a tool to hide all boulders not in stockpiles. that could be worked into this tool pretty easily, I think. new subcommands [un]hide, new condition: inside/outside stockpile

Reply:
Indeed, adding `[un]hide` should be easy. That's just a flag, isn't it? 

Regarding stockpiles: The use-case you outlined requires a (fast) test whether an item is in *any* stockpile. I'm not sure whats the best way to go about this.

Reply:
I think a scan of all stockpiles and a hashmap might be required. we have `dfhack.buildings.getStockpileContents(stockpile)` to get the list of stockpile contents, so iterating over all stockpiles and calling that would do it. It shouldn't be much worse than the existing iteration over all items.

Reply:
also have to iterate over the contents of containers returned by `getStockpileContents`

Reply:
That, and iterate over the contents of containers (i.e. bags) contained in containers. Seems to be working reasonably fast now, even without jumping through extra hoops.